### ISSUE TITLE
amdvlk does not work with DXVK

### ISSUE THREAD

**Issue by gardotd426** [doitsujin/dxvk#1286]
I wanted to test the performance of amdvlk with multiple games, so I installed it, as I knew it was supposedly able to live with RADV and able to be called with `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd64.json`. I tried to test it out, and it seems no game will launch with it. Doom using Proton (Steam and GE/Tk-Glitch builds) doesn't launch, no information why, even when running in a terminal. So I tried out launching my Titanfall 2 instance of Origin with `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd64.json`, and I get this:
```
info:  Required Vulkan extension VK_KHR_get_physical_device_properties2 not supported
info:  Required Vulkan extension VK_KHR_surface not supported
```
I know under common issues that that seems to be an issue with conflicting drivers with dedicated and integrated gpus, however I don't have an integrated GPU, just the RX 580. I also have seen someone suggest not having the icd loaders installed. I have `vulkan-icd-loader` and `lib32-vulkan-icd-loader` installed already. I was running `mesa-aco-git` from the valveaur repo, so I uninstalled it and installed all the individual components having it removes. Namely `lib32-libva-mesa-driver lib32-vulkan-radeon lib32-mesa lib32-mesa-vdpau libva-mesa-driver mesa mesa-vdpau opencl-mesa vulkan-mesa-layer vulkan-radeon`. My graphics card should absolutely be supported, and i checked to make sure that the amd_icd64.json and amd_icd32.json files were indeed in `/usr/share/vulkan/icd.d`, and they are. Every resource I've been able to find says that my card should be supported, I can have amdvlk and radv both and just dictate when I want to use amdvlk, and that it works with dxvk. However I can't get a single thing to even launch with it.

I don't know if this helps any, but when I tried to change amd_icd64.json to radeon_icd.x86_64.json, I still get the exact same logs with the exact same error. So that would mean it would be either a bug with dxvk or a bug with wine, right? When I run `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd64.json vulkaninfo`, I get no errors, and the extensions reported to be "missing" in the lutris logs are shown to be there. I've seen people say to do it with and without quotes around the path to the file, I tried both. And of course when i don't dictate anything, it loads radv and everything works fine.

I've tried this with: 
dxvk versions: 1.4.5, 1.4.6, 1.5, and a custom aur build of dxvk with the async patch, dxvk-mingw. With 1.4.5 and 1.4.6 Origin will still launch, but I still get all the same errors, plus the `DxvkInstance: Failed to create instance` errors, so obviously the games itself still won't work. With 1.5 and up it doesn't launch at all. 

### System information
- GPU: XFX GTS XXX RX 580 4GB
- Driver:Mesa 19.3 and Mesa-aco-git 20.0
- Wine version: lutris-nofshack-4.19, proton-tkg-5.0, proton-GE-4.20, tkg-protonified-3.21, staging 4.21
- DXVK version: 1.4.5, 1.4.6, 1.5
- Kernel both 5.4.3-2-MANJARO and 5.4.3-arch1-1-fsync
[Origin_d3d9.log](https://github.com/doitsujin/dxvk/files/3971361/Origin_d3d9.log)
[Origin_d3d11.log](https://github.com/doitsujin/dxvk/files/3971362/Origin_d3d11.log)
[Origin_dxgi.log](https://github.com/doitsujin/dxvk/files/3971363/Origin_dxgi.log)

**Comment by doitsujin** [comment]
This is not a bug, and DOOM uses Vulkan directly so this clearly isn't the right place to report this.

Anyway, if you set `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd64.json`, then you can **only** launch 64-bit applications since this is a 64-bit driver. Origin is a 32-bit app, so you additionally need to specify a path to a 32-bit driver (this could be RADV):
`VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd64.json:/usr/share/vulkan/icd.d/radeon_icd.i686.json`

**Comment by gardotd426** [comment]
Yes, thank you. I had tried it with some other games too, but they all ran Origin (I forgot about Doom being native vulkan, it's been a long night). I had tried to do exactly what you suggested, BUT I had used a comma instead of a colon. I just tried it with a colon, and Origin did start, however my card was still reported in the logs as `AMD RADV POLARIS10`, but I'm not sure if that means anything. I couldn't see anything on Resident Evil 2, by the time I got it to launch with any dxvk hud at all, I only had it set to 1, and it didn't show whether it was amdvlk or radv, and the next time I tried to boot, stupid drm told me I'd logged on from too many computers so now I have to wait until tomorrow. But it somewhat appears to be working, aside from still reporting the card as AMD RADV

**Comment by doitsujin** [comment]
If it reports cards as RADV then you're still using RADV.

**Comment by gardotd426** [comment]
I figured it out. I tested it out with a non-dxvk/windows application, I ran `./Geekbench-5.0.4-Linux/geekbench5 --compute-list`, which shows available rendering devices, both with the environment variable and without, and I got two different results (strangely enough when I run it without the variable, it lists 3 devices under Vulkan, AMD Radeon RX 580, RADV POLARIS10, and then AMD RADEON RX 580 again, but I think geekbench is just buggy with that). I went back into Lutris and changed the line you suggested, specifying both 32 and 64 bit, but made the second one `amd_icd32.json`. Now in Lutris where it used to say RADV, it now says AMD RX 580. I imagine if I had actually launched the game before, since it is a 64 bit game, it would have reported the RADV for Origin and the RX 580 for the actual game in the logs, but I'm not sure. Thanks for your help, I wouldn't have figured it out otherwise, every single thing I could find just said to export the one variable. I'll close this issue.
